src/backwardproblem.cpp

```diff
@@ -59,44 +65,55 @@ void BackwardProblem::workBackwardProblem() {
     }
 
     int it = model->nt() - 1;
-    model->initializeB(xB, dxB, xQB);
-    handleDataPointB(it);
-    solver->setupB(&which, model->getTimepoint(it), model, xB, dxB, xQB);
+    /* If we have posteq, infty timepoints were already treated */
+    for (int jt = model->nt() - 1; jt >= 0; jt--)
+        if (std::isinf(model->getTimepoint(it)))
```

Comment:
```suggestion
        if (std::isinf(model->getTimepoint(jt)))
```

?

Otherwise this loop does not make much sense, does it?